R/seq_linter.R

```diff
@@ -66,6 +76,15 @@ seq_linter <- function() {
     ]
   ")
 
+  map_funcs <-  xp_text_in_table(c("sapply", "lapply", "map"))
+  sequence_xpath <- glue("
+  //SYMBOL_FUNCTION_CALL[ { map_funcs } ]
+    /parent::expr/parent::expr[
+      preceding-sibling::expr/SYMBOL_FUNCTION_CALL[text() = 'unlist']
```

Comment:
nit: I would try and keep the logic in one direction on the AST: check sapply, then check seq_len, then check unlist, roughly

```
//SYMBOL_FUNCTION_CALL/parent::expr[following-sibling::expr/SYMBOL]/parent::expr/parent::expr[expr/SYMBOL_FUNCTION_CALL]
```

Also, please switch to use the new `find_function_calls` approach, which is appropriate here, e.g.

https://github.com/r-lib/lintr/blob/6ed78c2373ea435932151a5ae2090a3efc346e25/R/condition_call_linter.R#L83